src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -646,6 +646,52 @@ type internal TypeCheckInfo
         let items = items |> RemoveExplicitlySuppressed g
         items, nenv.DisplayEnv, m
 
+    /// Find union cases and compatible active patterns when the item under the cursor position is an identifier with the type of a DU.
+    let GetUnionCasesAndActivePatternsEnvironmentLookupResolutionsAtPosition cursorPos =
+        let rec doesActivePatternTakeTypeAsInput g ty paramType =
+            match paramType with
+            | TType_var (typar, _) ->
+                match typar.Solution with
+                | Some paramType ->
+                    doesActivePatternTakeTypeAsInput g ty paramType
```

Comment:
I'm concerned about subtyping and generic here.  Have you tested the case like this?
```fsharp
let (|AAAA|_|) (x: System.IComparable) = Some 1

let f () =
    match 3 with
    | AAAA s -> s + 1
    | n -> n + 2
```

Reply:
That won't be a problem since `3` is not a DU, so this specialized filtering does not kick in and `AAAA` is on the list.

What doesn't work is

![devenv_kKm0IabZDx](https://user-images.githubusercontent.com/5063478/167939409-d6487178-b36c-4181-a788-1e22fb57257a.png)

I'm using `typeEquiv`. Is there a function that could tell me if one type is a subtype of/assignable to another?

Reply:
If that's not possible, perhaps I could get rid of all this and simply show all active patterns? I think it's unlikely to see many of those, so it would make little difference.

Reply:
You can use this I believe:

```fsharp
                    if TypeRelations.TypeFeasiblySubsumesType 0 g amap m thisTy TypeRelations.CanCoerce ty then
```

Reply:
I think we ideally want to filter active patterns as well as we filter union cases.  It's not good long-term if tooling works better with union cases than active patterns as it encourages people to rely on the union cases (e.g. not hide them) and avoid using active patterns.